components/support/guid/src/lib.rs

```diff
@@ -97,6 +102,32 @@ impl FastGuid {
     }
 }
 
+impl Ord for FastGuid {
```

Comment:
Oh, yikes, nice catch with this bug.

That said, I'd prefer these be implemented on Guid and not FastGuid.

E.g. manually implement them all for Guid, deferring to `self.as_bytes()` (and not self.as_str() to avoid needing extra UTF-8 validation). That would also catch any cases where a guid that is small ends up in the non-FastGuid storage, not that that should ever happen.

Sorry, was taking a quick look through and missed this the first time.

Reply:
Not sure if I get this correct this time. Please take a look, thanks!